File: Models/GraphData.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace thing.Models
{
  public class DataPoint<T> {
    // Really a string, but we're doing date so we can work with it easier
    // public string name;
    public T name;
    public decimal value;
  }

  public class DataSet<T> {
    public string name;
    public List<DataPoint<T>> series;
  }
}

File: ClientApp/src/app/interfaces/graphdata.ts
/*
public class DataPoint {
    public string name;
    public decimal value;
  }

  public class DataSet {
    public string name;
    public List<DataPoint> series;
  }
  */

export interface DataPoint {
  // name: string;
  name: Date;
  value: number;
}

export interface DataSet {
    name: string;
    series: DataPoint[];
}